src/simplifyAST.js

```diff
@@ -2,7 +2,7 @@ function deepMerge(a, b) {
   Object.keys(b).forEach(function (key) {
     if (['fields', 'args'].indexOf(key) !== -1) return;
 
-    if (a[key] && b[key]) {
+    if (a[key] && b[key] && typeof a[key] === 'object') {
```

Comment:
Right, and that happens in the else statement. If it tries to deep merge b[key] as a string into an object it will likely still fail.